lib/client/doc.js

```diff
@@ -1077,7 +1088,10 @@ Doc.prototype._hardRollback = function(err) {
         inflightOp = null;
       }
 
-      if (!pendingOps.length) return;
+      if (!pendingOps.length) {
+        doc._isInHardRollback = false;
```

Comment:
If we want to avoid potential future issues with forgetting to reset `_isInHardRollback` on early returns, we could reset to false at the top of this `_fetch` callback